.github/workflows/tests.yaml

```diff
@@ -0,0 +1,59 @@
+name: Tests
+
+on:
+  workflow_run:
+    workflows: ['Code Checks']
+    types:
+      - completed
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm install
+
+  unit-tests:
+    needs: [setup]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Run Jest tests
+        uses: willcaul/jest-github-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
what secret is this

Reply:
It's automatic auth token for actions, required for this jest action. 

https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication

Reply:
can you name it as a token for Jest.. maybe JEST_GITHUB_TOKEN. also share the token with me on DM since i can't see it on github